js/blockTrackers.js

```diff
@@ -11,16 +11,29 @@ var entityList = JSON.parse(load.loadExtensionFile(entityListSource, 'json', 'ex
 
 var betterList = JSON.parse(load.loadExtensionFile('better-pages.txt', 'json'));
 
+var bg = chrome.extension.getBackgroundPage();
+var formerSocialBlocking = bg.isSocialBlockingEnabled;
+
 require.scopes.blockTrackers = (function() {    
     
     // If blocking option is enabled
     // and url matches a tracker pattern
     // block the request
-    function blockTrackers(url, currLocation) {
+    function blockTrackers(url, currLocation, tabId) {
         if (localStorage['blocking'] === 'true') {
             var host = extractHostFromURL(url);
+            var isWhiteListed = false;
+
+            if (formerSocialBlocking !== bg.isSocialBlockingEnabled) {
+                formerSocialBlocking = bg.isSocialBlockingEnabled;
+                trackers = load.processMozillaBlockList(blockList);
```

Comment:
As we saw while testing, onMessage is async and it's not the best place for remapping the list